pkg/volume/csi/csi_mounter.go

```diff
@@ -219,6 +219,27 @@ func (c *csiMountMgr) SetUpAt(dir string, fsGroup *int64) error {
 		return err
 	}
 
+	// apply volume ownership
+	if !c.readOnly && fsGroup != nil {
+		err := volume.SetVolumeOwnership(c, fsGroup)
+		if err != nil {
+			// attempt to rollback mount.
+			glog.Error(log("mounter.SetupAt failed to set volume ownership: %v", err))
+			glog.V(4).Info(log("mounter.SetupAt attempting to unpublish volume %s", c.volumeID))
+			if err := csi.NodeUnpublishVolume(ctx, c.volumeID, dir); err != nil {
+				glog.Errorf(log("mounter.SetupAt failed to unpublish [%s]: %v", c.volumeID, err))
+				return err
```

Comment:
this will only return the NodeUnpublishVolume err and not the original SetVolumeOwnership err although that was the original error that set this off, maybe we can return a combination of those errors in this case?

Reply:
@davidz627 Agreed, I will combine it with the original error